edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MetricsFileStorageTest.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Moq;
+    using Xunit;
+
+    public class MetricsFileStorageTest : IDisposable
+    {
+        DateTime fakeTime;
+        ISystemTime systemTime;
+        TempDirectory tempDirectory = new TempDirectory();
+
+        public MetricsFileStorageTest()
+        {
+            var systemTime = new Mock<ISystemTime>();
+            this.fakeTime = new DateTime(100000000);
+            systemTime.Setup(x => x.UtcNow).Returns(() => this.fakeTime);
+            this.systemTime = systemTime.Object;
+        }
+
+        [Fact]
+        public void Storage()
+        {
+            string directory = this.tempDirectory.GetTempDir();
```

Comment:
don't use tempDirectory; as you want to test that MetricsFileStorage will create the directory if it doesn't exist.

Reply:
used ```string directory = Path.Combine(this.tempDirectory.GetTempDir(), "metricsStore");``` since I still want the data to be deleted when done.